storm-client/src/jvm/org/apache/storm/grouping/LoadAwareShuffleGrouping.java

```diff
@@ -85,6 +85,7 @@ public void prepare(WorkerTopologyContext context, GlobalStreamId stream, List<I
         sourceNodeInfo = new NodeInfo(context.getThisWorkerHost(), Sets.newHashSet((long) context.getThisWorkerPort()));
         taskToNodePort = context.getTaskToNodePort();
         this.targetTasks = targetTasks;
+        capacity = targetTasks.size() == 1 ? 1 : Math.max(1000, targetTasks.size() * 5);
```

Comment:
Please add tests which cover the case that task count is beyond 1000 to see if it still guarantees correct distribution with acceptable margin.

Reply:
I just added the tests in the last commit. Is that what you were saying? Sorry if I misunderstood it

Reply:
@Ethanlm Yes that's what I meant. Thanks for addressing.